frontend/src/util.ts

```diff
@@ -70,4 +70,12 @@ function clusterDefaultUser(cluster: any) {
     "centos7": "centos"}[os]
 }
 
-export { getIn, swapIn, setIn, updateIn, findFirst, clusterDefaultUser }
+function clamp(num: number, min: number, max: number, step?: number): number
```

Comment:
issue(non-blocking): Since this function is a `util` and will likely be used throughout the codebase we should write a test for this. This would also help in document the `step` argument which isn't typically included in the clamp implementations I'm used to.

Otherwise, you could use [lodash.clamp](https://lodash.com/docs/4.17.15#clamp) since I believe we already have `lodash` among the dependencies (it does lack the step option though)